utils/modules/edge_utils.py

```diff
@@ -81,3 +130,135 @@ def faces_angle(normals, edges, pols):
             ang = 2*pi
         angles.append(ang)
     return angles
+
+def edges_normals_full(vertices, edges, faces):
+    '''Average of normals of the faces that share the edge (only the two first)'''
+    bm = bmesh_from_pydata(vertices, edges, faces, normal_update=True)
+    normals = [tuple(face.normal) for face in bm.faces]
+    bm.free()
+    vals = edges_normal(vertices, normals, edges, faces)
+
+    return vals
+
+def edges_normal(vertices, normals, edges, pols):
+    '''Average of normals of the faces that share the edge (only the two first)'''
```

Comment:
> It won't return a normal if faces are not flat.

It relies on bmesh capabilities to determinate the normals, I'm not sure how it deals with non-planar faces, but i just trusted in bmesh and didn't dig more on it...

>And what will be if an edge does not have adjacent faces?
>
>I think correct way will be get average normal from normals of adjacent vertices.

If there are no faces there are not valid vertex normals (for bmesh) .
I also thought (and coded) first the average or the vertex normals but it gave vertors that were not perpendicular to de edges (for example in the corner edges of a subdivided box) and as I see it the normals should be always perpendicular...

Reply:
>It relies on bmesh capabilities to determinate the normals, I'm not sure how it deals with non-planar faces, but i just trusted in bmesh and didn't dig more on it...

What do you mean? As far as I concern the `bmesh` module does not gives any normals for edges. You are create them according your own logic.

![no normals](https://user-images.githubusercontent.com/28003269/72682502-fffed000-3ae6-11ea-8dd1-eff684a02413.gif)
Your solution also does not gives perpendicular normals as I already mentioned.

Reply:
As you point the algorithm was created using my own logic, neither bmesh or blender deals with this concept if any software does (probably because the normal to a line should be a plane) . And it fails without polygons and with no-planar faces but works t to create this 
![image](https://user-images.githubusercontent.com/10011941/72688427-60056e80-3b07-11ea-8f44-577ef6cee979.png)
and from it to create this:
![image](https://user-images.githubusercontent.com/10011941/72688498-2719c980-3b08-11ea-830f-cf4a1fd6e933.png)

If you can improve the algorithm in any way covering the missing cases or improving its logic you are very welcome to do it. I was thinking in it for a while and it this is  how far I could arrive

Reply:
It looks working correctly:
```Python
y = (edge.verts[1].co - edge.verts[0].co).normalized()
_normal = (edge.verts[0].normal + edge.verts[1].normal).normalized()
x = y.cross(_normal)
normal = x.cross(y)
```

Reply:
Except for the fact that bm.edges does not respect the order of the input your function works better and is simpler, I will implement it. 
Thanks!